### PR TITLE
chore: Use Github actions for CI.

### PR DESCRIPTION
**Pull request by kinyoklion**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: Build and Test
+
+on:
+  push:
+    branches: [main, rlamb/github-action-ci]
```
Comment (kinyoklion) [inline .github/workflows/ci.yml:5]: Temporarily run on this branch for testing.

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1 @@
+* text=auto eol=lf
```
Comment (kinyoklion) [inline .gitattributes:1]: Git converting lint endings was causing problem with linting. `LF` is totally fine on modern windows.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Build
+        run: npm run build
+      - name: Test
+        run: npm test
+        env:
+          JEST_JUNIT_OUTPUT_FILE: "reports/junit/js-test-results.xml"
+      - name: Lint
+        run: npm run lint
+      - name: Test Report
+        uses: dorny/test-reporter@v1
```
Comment (kinyoklion) [inline .github/workflows/ci.yml:42]: Totally digging this: https://github.com/launchdarkly/openfeature-node-server/actions/runs/5425662751/jobs/9866718683

File: tsconfig.eslint.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
   "extends": "./tsconfig.json",
-  "include": ["/**/*.ts"],
+  "include": ["**/*.ts"],
```
Comment (kinyoklion) [inline tsconfig.eslint.json:3]: Linting was not working on windows, so all the `./` removal is to appease windows.

File: yarn.lock
Comment (kinyoklion) [inline yarn.lock:1]: Adding the yarn lock.

**Comment by kinyoklion** [comment]
Ben removed the circleci build, so that should go away.

**Review by louis-launchdarkly — APPROVED** [review]

**Pull request merged**